src/components/data-exploration/differential-expression-tool/AdvancedFilteringModal.jsx

```diff
@@ -0,0 +1,162 @@
+    pct1: [0, 100],
+    pct2: [0, 100],
+    logfc: [-50, 50],
+    auc: [0, 1],
+    pValue: [0, 1],
+  };
+
+  const presetFilters = {
+    'Up-regulated': {
+      criteria: 'logfc',
+      condition: 'gt',
+      value: 0,
+    },
+    'Down-regulated': {
+      criteria: 'logfc',
+      condition: 'lt',
+      value: 0,
+    },
+    Significant: {
+      criteria: 'pValue',
+      condition: 'lt',
+      value: 0.05,
+    },
+  };
+  const renderPresetFilters = (add) => (
+    <Menu
+      onClick={(e) => {
+        add(presetFilters[e.key]);
+        changeCriteria(presetFilters[e.key].criteria);
+      }}
+    >
+      {Object.keys(presetFilters).map((filter) => (
+        <Menu.Item key={filter}>
+          {filter}
+        </Menu.Item>
+      ))}
+    </Menu>
+  );
+
+  const changeCriteria = (criteria, index = false) => {
+    let currentIndex = index;
+    if (index === false) {
+      const { filterForm } = form.getFieldsValue();
+      currentIndex = filterForm.length - 1;
+    }
+    const currentDomains = valuesDomain;
+    currentDomains[currentIndex] = valueRestrictions[criteria];
+    setValuesDomain(currentDomains);
+    // without the next line the form does not re-render, so the max and min values are not updated
+    form.setFieldsValue({});
+  };
+
+  return (
+    <Modal
+      visible
+      title='Advanced filters'
+      onCancel={onCancel}
+      okText='Apply filters'
+    >
+      <Form form={form}>
+
+        <Form.List
+          name='filterForm'
+        >
+          {(fields, { add, remove }) => (
+            <>
+              {fields.map((field, index) => (
+                <Space key={field.key} align='baseline'>
+                  <Space direction='horizontal'>
+                    <Form.Item
+                      name={[field.name, 'criteria']}
+                      fieldKey={[field.criteria, 'criteria']}
+                    >
+                      <Select
+                        placeholder='Select criteria'
+                        style={{ width: 140 }}
+                        onChange={(value) => changeCriteria(value, index)}
+                        options={criteriaOptions}
+                      />
+                    </Form.Item>
+
+                    <Form.Item
+                      name={[field.name, 'condition']}
+                      fieldKey={[field.condition, 'condition']}
+                    >
+                      <Select
+                        placeholder='Select condition'
+                        options={conditionOptions}
+                        style={{ width: 150 }}
+                      />
+                    </Form.Item>
+
+                    <Form.Item
+                      name={[field.name, 'value']}
+                      fieldKey={[field.value, 'condition']}
+                    >
+                      <InputNumber
+                        style={{ width: 140 }}
+                        min={valuesDomain[index] ? valuesDomain[index][0] : 0}
```

Comment:
couldn't you just get the min from `valueRestrictions[field.key]` instead?

Reply:
would this allow you to get rid of `valuesDomain`? Or are you using it for something different?

Reply:
When the form is empty fields is empty and this will crash

Reply:
I didn't understand the empty fields causing a crash thing.

On the max and min I understand that they have to change, but I don't see why accessing the criteria directly will not work, right now you are using `form.setFieldsValue({});` to trigger a re render, right? So it would still re render.

We can have a call tomorrow if you want to talk about this, perhaps I am missing something.

Reply:
As an apart from everything above, if you want to think of an alternative to having to re render to update the min and max we could maybe implement a wrapper component for `InputNumber` that allows us to have `min` and `max` be function calls.

The benefit of a function is that it would allow us to calculate min and max when we need to and not only on render. For example, we could have the wrapper store the `min` and `max` by index and then check against them when `onChange` is triggered to set the value we actually want to be there.

Reply:
The re-render occurs when you change something so it is not slowing down. I'll leave it like this for now, because I don't think the change will make a noticable difference in experience.